samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -61,17 +65,24 @@
   private final CoordinationUtils coordinationUtils;
 
   private JobModel newJobModel;
-  private String newJobModelVersion;  // version published in ZK (by the leader)
```

Comment:
I refactored it because it didn't need to be a member variable. However, since I don't have a good idea about this code, I think I will change this back. 

> Is it relevant to the processorId change?
No. Not related at all.